### PR TITLE
chore: add utility function to check eip1559 compliance

### PR DESCRIPTION
**Pull request by leoslr**
## Description of the changes

Small refacto to regroup logic related to eip1559 compliance
[Asana task](https://app.asana.com/0/1203350802794948/1203600402430065)

### CODE REVIEW

File: packages/utils/src/providers.ts [hunk 1]
```diff
@@ -147,9 +149,22 @@ const getCeloProvider = (): providers.Provider => {
...
 };
 
+const isEip1559Supported = async (provider: any, logger: LogTypes.ILogger): Promise<boolean> => {
+  try {
+    await provider.send('eth_feeHistory', [1, 'latest', []]);
+    return true;
+  } catch (e) {
+    logger.warn(
+      'This RPC provider does not support the "eth_feeHistory" method: switching to legacy gas price',
+    );
```
Comment (benjlevesque) [inline packages/utils/src/providers.ts:159]: I think this log should be removed from the utility

File: packages/utils/src/providers.ts [hunk 2]
```diff
@@ -147,9 +149,22 @@ const getCeloProvider = (): providers.Provider => {
   return provider;
 };
 
+const isEip1559Supported = async (provider: any, logger: LogTypes.ILogger): Promise<boolean> => {
```
[the remark this answers is not in the document]
Comment (benjlevesque) [reply packages/utils/src/providers.ts:152]: a hack you can do is to type it as `providers.Provider`, then cast it to `JsonRpcProvider`.

Not great, but still better than `any` ;)

**Review by benjlevesque — APPROVED** [review]

**Review by alexandre-abrioux — APPROVED** [review]
nice!

**Comment by leoslr** [comment]
@benjlevesque this does not work. As soon as the type of the parameter is related to the `ethers` library it raise the error, it's related to how we export functions in the `utils` package.

**Comment by benjlevesque** [comment]
> @benjlevesque this does not work. As soon as the type of the parameter is related to the `ethers` library it raise the error, it's related to how we export functions in the `utils` package.

OK, so maybe https://github.com/RequestNetwork/requestNetwork/pull/1023 will solve it...

**Review by yomarion — APPROVED** [review]

File: packages/utils/src/providers.ts (hunk 1 above)
Comment (leoslr) [reply packages/utils/src/providers.ts]: I kept it there but made it optional.

**Pull request merged**